src/Agent.Plugins/RepositoryPlugin.cs

```diff
@@ -190,6 +191,17 @@ public override async Task RunAsync(AgentTaskPluginExecutionContext executionCon
                 repo.Properties.Set<string>(Pipelines.RepositoryPropertyNames.Path, expectRepoPath);
             }
 
+            if (!PlatformUtil.RunningOnWindows && string.Equals(repo.Type, Pipelines.RepositoryTypes.Tfvc, StringComparison.OrdinalIgnoreCase))
+            {
+                TeeUtil.DownloadTeeIfAbsent(
```

Comment:
There is a lot of duplicate code here if you look at [another DownloadTeeIfAbsent call](https://github.com/microsoft/azure-pipelines-agent/pull/3684/files#diff-09c6c4d5b7424b6e09847842d5f3897d37cdde5577316b13a564c4e513a901d2R221).

I was pondering if we should use polymorphism and pass execution context to this method instead. Unfortunately, `AgentTaskPluginExecutionContext` and `IExecutionContext` have no common interfaces, so this won't work.

Reply:
Outdated